winedarksea/AutoTS
# 0.4.0

# Latest
* Note: the plan is to replace ZeroesNaive model with ConstantNaive in a future release
* fix bug where score was failing to generate if made_weighting > 0 and forecast_length = 1
* made MADE functional on forecast_length=1 if df_train is provided
* SMAPE now in __repr__ of fit AutoTS class
* contour now works on forecast_length = 1
* Added NeuralProphet model
* made the probabilistic modeling of MultivariateRegression a parameter which only occurs when 'deep' mode is active (too slow)
* added more params to pass through to Prophet
* add phi damping to Detrend transformer
* add window slice to Detrend transformer
* added 'simple_2" datepart method
* added package to conda-forge
* preclean method added to AutoTS
* added median and midhinge point_methods to BestN Ensembles
* added additional model selections to 'simple' and 'subsample' ensemble
* switch LightGBM back to MultiOutputRegressor from RegressorChain for speed (with LightGBMRegressorChain replacing)
* removed top-level datasets `requests` dependency
* Added EWMAFilter
* improved NaN in forecast check
* added fail_on_forecast_nan (bool) to AutoTS.predict and model_forecast, if False, can now allow forecasts with NaN to be returned
* add return_model to model_forecast for model and transformer
* fixed bug where Detrend failing with non-datetime index
* improved error handling in Transformers to explicitly reference which failed
* added random.seed() setting in AutoTS which actually seems to standardize the runs
* sped up assembling/concat of horizontal ensembles for large numbers of series
* added polynomial_degree to date_part (and ~Transformer and ~Regression)
* updated infer_frequency and utilized in model base class
* added additional datasets (analytics.gov and severe weather) to load_lively_daily and modified pytrends load
* added rps to metrics (although no plans to build it into evaluation)
* added 'Ridge' and 'GaussianProcessRegressor' as model options for Regressions
* enforcing consistency on inner n_jobs with MultiOutputRegressor
* add DynamicFactorMQ model from Statsmodels
* added plot_per_series_smape and list_failed_model_types to output more run information from AutoTS class
* increased number of best per series models added to models to validate (models to validate has become more of a baseline than a firm number)
* finally transitioned `ensemble` parameter fully to a list from the original comma-sep string list
* MLE and iMLE logarithmic metrics for targeting under- and over- estimation
* MAGE metric for error on rollup forecasts
* Mosaic ensembles now include a metric_weighting variation including MAE, RMSE and SPL weighting (abs error, square error, pl error) (unscaled)
* minor but noticeable speedups to TemplateWizard and inferred_normal functions
* added EventRiskForecast for determing risk of exceeding limits (should be considered in beta for now)
* back_forecast now has tail/eval_periods configured
* changed behavior of import_template, by default simple ensembles are unpacked but no longer included in template unless include_ensemble is True.
